### ISSUE TITLE
Failed "typing_copilot init" due to mypy crash during loose config run

### ISSUE THREAD

**Issue by obi1kenobi** [obi1kenobi/typing-copilot#1]
Running `typing_copilot init` on the `arviz` [project repo](https://github.com/arviz-devs/arviz) causes a `mypy` crash during the loose config `mypy` run. Here's the `mypy` crash reproduced with `mypy`'s most recent `master` branch:
```
$ mypy --config-file ./custom_mypy.ini -p arviz --show-traceback
arviz/rcparams.py:286: error: Cannot determine consistent method resolution order (MRO) for "RcParams"
/my/code/path/arviz/venv/lib/python3.8/site-packages/xarray/core/dataarray.py:2040: error: INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.rtfd.io/en/latest/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.790+dev.0b4a2c9d8edb57e2b61219c96eefe010073a5c14
Traceback (most recent call last):
  File "/my/code/path/arviz/venv/bin/mypy", line 8, in <module>
    sys.exit(console_entry())
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/__main__.py", line 11, in console_entry
    main(None, sys.stdout, sys.stderr)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/main.py", line 90, in main
    res = build.build(sources, options, None, flush_errors, fscache, stdout, stderr)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/build.py", line 180, in build
    result = _build(
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/build.py", line 254, in _build
    graph = dispatch(sources, manager, stdout)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/build.py", line 2630, in dispatch
    process_graph(graph, manager)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/build.py", line 2953, in process_graph
    process_stale_scc(graph, scc, manager)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/build.py", line 3051, in process_stale_scc
    graph[id].type_check_first_pass()
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/build.py", line 2111, in type_check_first_pass
    self.type_checker().check_first_pass()
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 294, in check_first_pass
    self.accept(d)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 401, in accept
    stmt.accept(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/nodes.py", line 940, in accept
    return visitor.visit_class_def(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 1723, in visit_class_def
    self.accept(defn.defs)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 401, in accept
    stmt.accept(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/nodes.py", line 1005, in accept
    return visitor.visit_block(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 1972, in visit_block
    self.accept(s)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 401, in accept
    stmt.accept(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/nodes.py", line 677, in accept
    return visitor.visit_func_def(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 726, in visit_func_def
    self._visit_func_def(defn)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 730, in _visit_func_def
    self.check_func_item(defn, name=defn.name)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 792, in check_func_item
    self.check_func_def(defn, typ, name)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 975, in check_func_def
    self.accept(item.body)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 401, in accept
    stmt.accept(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/nodes.py", line 1005, in accept
    return visitor.visit_block(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 1972, in visit_block
    self.accept(s)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 401, in accept
    stmt.accept(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/nodes.py", line 1196, in accept
    return visitor.visit_if_stmt(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 3255, in visit_if_stmt
    self.accept(b)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 401, in accept
    stmt.accept(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/nodes.py", line 1005, in accept
    return visitor.visit_block(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 1972, in visit_block
    self.accept(s)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 401, in accept
    stmt.accept(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/nodes.py", line 1130, in accept
    return visitor.visit_for_stmt(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 3478, in visit_for_stmt
    iterator_type, item_type = self.analyze_iterable_item_type(s.expr)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checker.py", line 3511, in analyze_iterable_item_type
    return iterator, echk.check_method_call_by_name(nextmethod, iterator, [], [], expr)[0]
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checkexpr.py", line 2346, in check_method_call_by_name
    method_type = analyze_member_access(method, base_type, context, False, False, True,
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checkmember.py", line 126, in analyze_member_access
    result = _analyze_member_access(name, typ, mx, override_info)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checkmember.py", line 143, in _analyze_member_access
    return analyze_instance_member_access(name, typ, mx, override_info)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checkmember.py", line 225, in analyze_instance_member_access
    return analyze_member_var_access(name, typ, info, mx)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checkmember.py", line 379, in analyze_member_var_access
    return analyze_var(name, v, itype, info, mx, implicit=implicit)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/checkmember.py", line 581, in analyze_var
    dispatched_type = meet.meet_types(mx.original_type, itype)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/meet.py", line 50, in meet_types
    return t.accept(TypeMeetVisitor(s))
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/types.py", line 833, in accept
    return visitor.visit_instance(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/meet.py", line 497, in visit_instance
    args.append(self.meet(ta, sia))
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/meet.py", line 647, in meet
    return meet_types(s, t)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/meet.py", line 50, in meet_types
    return t.accept(TypeMeetVisitor(s))
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/types.py", line 1386, in accept
    return visitor.visit_tuple_type(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/meet.py", line 584, in visit_tuple_type
    return TupleType(items, tuple_fallback(t))
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/typeops.py", line 44, in tuple_fallback
    return Instance(info, [join_type_list(typ.items)])
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/join.py", line 531, in join_type_list
    joined = join_types(joined, t)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/join.py", line 105, in join_types
    return t.accept(TypeJoinVisitor(s))
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/types.py", line 833, in accept
    return visitor.visit_instance(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/join.py", line 158, in visit_instance
    nominal = join_instances(t, self.s)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/join.py", line 371, in join_instances
    elif t.type.bases and is_subtype_ignoring_tvars(t, s):
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/subtypes.py", line 143, in is_subtype_ignoring_tvars
    return is_subtype(left, right, ignore_type_params=True)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/subtypes.py", line 93, in is_subtype
    return _is_subtype(left, right,
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/subtypes.py", line 135, in _is_subtype
    return left.accept(SubtypeVisitor(orig_right,
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/types.py", line 833, in accept
    return visitor.visit_instance(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/subtypes.py", line 264, in visit_instance
    if right.type.is_protocol and is_protocol_implementation(left, right):
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/subtypes.py", line 550, in is_protocol_implementation
    is_compat = is_subtype(subtype, supertype, ignore_pos_arg_names=ignore_names)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/subtypes.py", line 93, in is_subtype
    return _is_subtype(left, right,
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/subtypes.py", line 135, in _is_subtype
    return left.accept(SubtypeVisitor(orig_right,
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/types.py", line 1825, in accept
    return visitor.visit_partial_type(self)
  File "/my/code/path/arviz/venv/lib/python3.8/site-packages/mypy/subtypes.py", line 468, in visit_partial_type
    raise RuntimeError
RuntimeError: 
/my/code/path/arviz/venv/lib/python3.8/site-packages/xarray/core/dataarray.py:2040: : note: use --pdb to drop into pdb
```

Interestingly, the `mypy` crash happens with our loose `mypy` config, but not with `mypy`'s default config. Our loose config is as follows:
```
[mypy]
no_implicit_optional = False
strict_optional = False
warn_redundant_casts = False
check_untyped_defs = False
disallow_untyped_calls = False
disallow_incomplete_defs = False
disallow_untyped_defs = False
disallow_untyped_decorators = False
ignore_missing_imports = True
```
To get `mypy`'s default config, simply remove the `--config-file ./custom_mypy.ini` from the `mypy` invocation.

Since `mypy` crashes completely on this, its output cannot be parsed by `typing_copilot`. This is preventing it from producing actionable advice and enabling its adoption into the `arviz` project.

Thanks to @ColCarroll for trying out `typing_copilot` on `arviz` and pointing out the `mypy` issue.

**Comment by obi1kenobi** [comment]
Underlying issue: https://github.com/python/mypy/issues/9437

**Comment by obi1kenobi** [comment]
Fixed in 0.5.0.
